src/common/hist_util.cu

```diff
@@ -267,9 +269,10 @@ void ProcessWeightedBatch(int device, const SparsePage& page,
                              batch_it, dummy_is_valid,
                              0, sorted_entries.size(),
                              &cuts_ptr, &column_sizes_scan);
-
-  auto const& h_cuts_ptr = cuts_ptr.ConstHostVector();
   auto d_cuts_ptr = cuts_ptr.DeviceSpan();
+  detail::RemoveDuplicatedCategories(device, info, d_cuts_ptr, &sorted_entries,
```

Comment:
I don't see a unit test for `RemoveDuplicatedCategories`. Can you write on?

Reply:
It's not exposed in header.  Does `TestCategoricalSketch` suffice?

Reply:
Maybe. I'd like to understand how `TestCategoricalSketch` is passing. See my comment below.